programs/survival/graves.sc

```diff
@@ -0,0 +1,419 @@
+    if(!ownership && !global_settings:'allowOnlyOwner', _not_owner_clicked(player,block));
+);
+
+__on_player_breaks_block(player, block)->(
+
+    if(block != 'player_head' || player~'gamemode'!='survival', return());
+    nbt = nbt(block_data(block));
+
+    //Replace the broken player head
+    schedule(0,_(outer(player),outer(nbt)) -> (
+        if(player~'gamemode'!='survival', return());
+        run(str('setblock %s %s %s minecraft:player_head{ExtraType:"%s"}',nbt:'x',nbt:'y',nbt:'z',nbt:'SkullOwner':'Name'));
+    ));
+
+    //Delete the player head item that will drop as entity
+    schedule(0, _(outer(player),outer(pos(block))) -> (
+        
+        coords = pos(block);
+        //Found 2 to be reliable experimentally, needs more testing
+        dx = 2;
+        dy = 2;
+        dz = 2; 
+        if(query(player,'gamemode')=='creative',return());
+        selector_item = str('@e[type=item,x=%d,y=%d,z=%d,dx=%d,dy=%d,dz=%d]',coords:0,coords:1,coords:2,dx,dy,dz);
+        item = entity_selector(selector_item); 
+        selector_xp = str('@e[type=experience_orb,x=%d,y=%d,z=%d,dx=%d,dy=%d,dz=%d]',coords:0,coords:1,coords:2,dx,dy,dz);
+        xp = entity_selector(selector_xp);
+        if(length(item)!=0, modify(entity_selector(selector_item):0,'remove'));    
+        if(length(xp)!=0, modify(entity_selector(selector_xp):0,'remove'));
+
+    ));
+);
+
+__on_explosion_outcome(pos, power, source, causer, mode, fire, blocks, entities)->(
+    for(blocks,
+        if(_ == 'player_head',
+            posG = pos(_);
+            posG:0 = floor(posG:0);
+            posG:1 = floor(posG:1);
+            posG:2 = floor(posG:2);
+            schedule(50, _(outer(posG)) -> (
+                set(posG,'air');
+                run(str('setblock %s %s %s minecraft:player_head{ExtraType:"%s"}',posG:0,posG:1,posG:2,global_gdata:str(posG):'name'));
+            ));
+        ))
+);
+
+__on_player_dies(player)->(
+
+    if(!inventory_has_items(player), return());
+
+    //To do: Nearest air block
+    if(air(pos(player)),
+        gravepos = pos(player),
+        //else
+        gravepos = pos_offset(pos(player),'up');
+    );
+
+    gravepos:0 = floor(gravepos:0);
+    gravepos:2 = floor(gravepos:2);
+    if(pos(player):1<0, gravepos:1 = top('motion',gravepos:0,0,gravepos:2) + 1); 
+
+    print(player,format(str('nb A grave has been formed at %d, %d, %d in %s',gravepos:0,gravepos:1,gravepos:2,player~'dimension')));
+    
+    // Make the grave
+    set(gravepos,'air');
+    run(str('setblock %s %s %s minecraft:player_head{ExtraType:"%s"}',gravepos:0,gravepos:1,gravepos:2,player));
+
+    item_count=0;
+    global_gdata:str(gravepos)={};
+
+    c_for(i=0,i<inventory_size(player),i+=1,
+
+        if(inventory_get(player,i)!=null,
+
+            slotItem= inventory_get(player,i);
+            global_gdata:str(gravepos):str(i) = slotItem;
+            item_count+=1;
+            _saveGData();
+        
+        );
```

Comment:
Couldn't you do this as a list? Like wouldn't that make it easier to iterate through?

Reply:
I did not do this as a list because the grave auto-pickup by owner feature depends on being able to know the exact position of each item in inventory. I had to add the inventory==null check just to get around the issue of a very big save file